framework/src/engine/generator/generator.ts

```diff
@@ -622,10 +625,13 @@ export class Generator {
 	}
 
 	private _handleFinalizedHeightChanged(from: number, to: number): Promise<void>[] {
+		if (from >= to) {
+			return [];
+		}
 		const promises = [];
 		const stateStore = new StateStore(this._blockchainDB);
 		for (const [address, pairs] of this._keypairs.entries()) {
-			for (let height = from + 1; height <= to; height += 1) {
+			for (let height = from + 1; height < to; height += 1) {
```

Comment:
According to this in the LIP
`h1 = getMaxRemovalHeight() and h2 = getBFTHeights().maxHeightPrecommitted`

here we are starting with `getMaxRemovalHeight()+2` to `h2 - 1 `as we increment it above at Line206. Shouldn't this be?
```suggestion
			for (let height = from; height < to; height += 1) {
```

Reply:
ah that's true. the loop start from +1.
I think the input to this function needs to be changed because when it's called when `maxHeightPrecommited` is changed, we still want to certify from last `maxHeightPrecommited + 1` to current `maxHeightPrecommited`

Reply:
@janhack this needs to be updated in the [LIP](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0061.md#initial-single-commit-creation) as well?

Reply:
So there are two cases when this function is called:

Case 1: A block is processed and `maxHeightPrecommitted` changes from `h1` to `h2`, see https://github.com/LiskHQ/lips/blob/main/proposals/lip-0061.md#creation-after-block-processing where `h1 == from` and `h2 == to`. In this case, we want to iterate from `h1 + 1` to `h2` so the current version of the PR is correct (could you point me though where the function is called in this case?).

Case 2: The node restarts and we want to re-create single commits, see https://github.com/LiskHQ/lips/blob/main/proposals/lip-0061.md#initial-single-commit-creation where `maxHeightCertified == from` and `maxHeightPrecommitted == to` (the LIP needs an update here as described above). I also think the PR is correct here and I don't see how you get  `+ 2` @ishantiw .

Reply:
@janhack the code is updated now in this commit https://github.com/LiskHQ/lisk-sdk/pull/8095/commits/e1ce2650c28431b494f95c0f650ecec59dbd5a0c but earlier we had 
`this._handleFinalizedHeightChanged(maxHeightCertified + 1, maxHeightPrecommitted)`
and the loop is starting `for (let height = from + 1; height < to; height += 1) {` so we were getting `+2`